src/DynamoCoreWpf/UI/GuidedTour/GuidesValidationMethods.cs

```diff
@@ -93,6 +94,8 @@ internal static void ExecuteTermsOfServiceFlow(Step stepInfo, StepUIAutomation u
                     if (element != null)
                         ManageEventHandler(element, handler.HandlerElementEvent, handler.ExecuteMethod);
                 }
+
+                ownedWindow.Closed += CloseGuidedTour;
```

Comment:
Let's also unsubscribe it later

Reply:
Make sense because the TOU window is not a singleton right?

Reply:
It is not, but only once instance will be created because of the workflow we have now.